### ISSUE TITLE
Twitter/OAuth Provider successful login callback not being not received

### ISSUE THREAD

**Issue by Sagar-AppLogics** [firebase/firebase-ios-sdk#4415]
Environment:

Xcode version: 11.2.1
Firebase SDK version: 6.13.0
Firebase Component: Auth
Component version: 6.4.0

Problem:
Twitter/OAuth Provider successful login callback not being not received

Recreate by:
Create a new iOS Application project using SceneDelegate.
Setup twitter authentication using the following guide:
https://firebase.google.com/docs/auth/ios/twitter-login?authuser=0

@rizafran your fix for the issue https://github.com/firebase/firebase-ios-sdk/issues/4414 sheds light on a new issue:

I am being directed through the login process etc, but once the login is complete, I am not successfully redirected back to the application - I am forced to close the dialog manually which results in a cancellation from the completion handler.

I noticed the same issue with Facebook authentication in iOS 13 applications that implement a SceneDelegate - see my post here: https://github.com/jeduan/cordova-plugin-facebook4/issues/826#issuecomment-559630090

In Apps that do not implemented a SceneDelegate, AppDelegate handle the completion via `func application(_ application: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any] = [:]) -> Bool` however since the introduction of the SceneDelegate, this is now as you would expect - deferred to the SceneDelegate, namely the following method is now fired `func scene(_ scene: UIScene, openURLContexts URLContexts: Set<UIOpenURLContext>)`. I checked and this was indeed fired immediately after successful login to Twitter.

So this leads me to believe that the issue now is that the returned call is not being handled and therefore the login is not completing as expected.

Maybe there is a way to manually handle the response for the twitter login, in the way the facebook login is handled. Here is my solution for handling the Facebook Login:

```
func scene(_ scene: UIScene, openURLContexts URLContexts: Set<UIOpenURLContext>) {
    guard let url = URLContexts.first?.url else {
        return
    }
    let _ = ApplicationDelegate.shared.application(
        UIApplication.shared,
        open: url,
        sourceApplication: nil,
        annotation: [UIApplication.OpenURLOptionsKey.annotation])        
}
```

_Originally posted by @Sagar-AppLogics in https://github.com/firebase/firebase-ios-sdk/issues/4414#issuecomment-559841427_

**Comment by Sagar-AppLogics** [comment]
So after a (short) but of digging, I have come across the file: FirebaseAuth/FIRAuth.h which contains the function `- (BOOL)canHandleURL:(nonnull NSURL *)URL;` or `canHandle(URL: URL:)` at line 785.

Looking at the comments from lines: 775 to 784 - specifically the remark from line 782 to 783:

> @remarks If swizzling is disabled, URLs received by the application delegate must be forwarded
>     to this method for phone number auth to work.

This makes sense since these URLs are usually handled without further intervention with regards to applications that do not implement a SceneDelegate. so I tried the following and it worked:

```
func scene(_ scene: UIScene, openURLContexts URLContexts: Set<UIOpenURLContext>) {
    for URLContext in URLContexts {
        let url = URLContext.url
        Auth.auth().canHandle(url)
    }
}
```

It would be good to introduce swizzling into the SceneDelegates methods somehow, I guess one could also argue that it would be useful for developers to better understand the implications and changes when implementing a SceneDelegate - but I guess that was Apple's job to highlight better.

**Comment by rizafran** [comment]
Hi @Sagar-AppLogics, thanks for filing a separate issue and for sharing your findings.

Looks like the issue you're having here is related to #3706 which shows a blank page and not redirecting back to the app. The workaround here is to implement the last code snippet on this [link](https://firebase.google.com/docs/auth/ios/phone-auth#appendix:-using-phone-sign-in-without-swizzling) which you have also mentioned that it worked on your end.

**Comment by renkelvin** [comment]
The latest release (FirebaseAuth 6.4.1, GoogleUtilities 6.4.0) should fix this issue.
